sql-plugin/src/main/311until320-nondb/scala/com/nvidia/spark/rapids/shims/Spark31XShims.scala

```diff
@@ -279,9 +279,14 @@ abstract class Spark31XShims extends Spark301until320Shims with Logging {
     }),
     GpuOverrides.expr[GetMapValue](
       "Gets Value from a Map based on a key",
-      ExprChecks.binaryProject(TypeSig.STRING, TypeSig.all,
-        ("map", TypeSig.MAP.nested(TypeSig.STRING), TypeSig.MAP.nested(TypeSig.all)),
-        ("key", TypeSig.lit(TypeEnum.STRING), TypeSig.all)),
+      ExprChecks.binaryProject(
+        (TypeSig.commonCudfTypes + TypeSig.ARRAY + TypeSig.STRUCT + TypeSig.NULL +
+          TypeSig.DECIMAL_128 + TypeSig.MAP).nested(),
```

Comment:
The docs were not updated, but from what is listed here we support getting values from a map that are themselves maps? I don't see any tests for this. I see structs, but they only go to a single level. I didn't review the C++ code so I don't know exactly what is supported for values so I don't know if this is a bug here or it is a missing test.

Reply:
I can add tests for these combinations.

Reply:
One or two of them would be good so that we know it is covered. You don't need to go crazy with it though.

Reply:
Done now.